pmd-java/src/main/java/net/sourceforge/pmd/lang/java/oom/ParameterizedMetricKey.java

```diff
@@ -35,25 +35,13 @@ private ParameterizedMetricKey(MetricKey<? extends Metric> key, MetricVersion ve
 
     @Override
     public String toString() {
-        return "ParameterizedMetricKey{key=" + key + ", version=" + version + '}';
+        return "ParameterizedMetricKey{key=" + key.name() + ", version=" + version.name() + '}';
     }
 
 
     @Override
     public boolean equals(Object o) {
-        if (this == o) {
-            return true;
-        }
-        if (o == null || getClass() != o.getClass()) {
-            return false;
-        }
-
-        ParameterizedMetricKey that = (ParameterizedMetricKey) o;
-
-        if (!key.equals(that.key)) {
-            return false;
-        }
-        return version.equals(that.version);
+        return this == o;
```

Comment:
Right, this can be done, since we have a factory method.
Please double check, that we do not run into issues, if we serialize/deserialize instances of this class - Object deserialization would be the last possibility, where new instances are created, that this class doesn't control (though could be solved by "readResolve" implementation).